### DISCUSSION TITLE
Confused by the terminology: backup vs snapshot

### DISCUSSION THREAD

**Discussion by ovizii** [garethgeorge/backrest#343] [Q&A] [answered]
I did read the docs: https://garethgeorge.github.io/backrest/introduction/getting-started#terminology 

but I am still wondering why some of my plans show backups and snapshots, some just snapshots?

It is easier to show then describe:

Only snapshots and one scheduled backup:

![image](https://github.com/garethgeorge/backrest/assets/837902/117222ed-4a9e-42f0-92db-68d7cccbfbe8)

Snapshots and backups mixed:

![image](https://github.com/garethgeorge/backrest/assets/837902/8f073dfc-2824-4fb7-9dea-c040ae5b1b79)

I am just trying to understand. A backup seems to always consists of a snapshot and a backup and offers more info:

![image](https://github.com/garethgeorge/backrest/assets/837902/7a28905b-c898-4ae8-beaa-b9d04304579a)

Compared to a simple snapshot which offers not info whatsoever.

**Comment by garethgeorge** [comment]
A backup corresponds to an invocation of the `restic backup` command, it's a backrest concept that captures a command run on your behalf and the output of that command. The same goes for forget, prune, check, etc entries that you will see in the UI. In the tree view, "backups" are collections of operations grouped by the snapshot ID they're associated with (e.g. an execution of a plan you've configured).

The snapshot record is a restic snapshot, in your repo, indexed by backrest. It has fewer details because restic doesn't actually keep track of the ifles processed, bytes added, and other metrics. That's all data that's tracked and stored by backrest. 

You can click `Run Command` and type `snapshots` to see the snapshots in your repo directly if you're wanting to understand more about the underlying repo and what backrest is creating in there for you.

**Comment by ovizii** [comment, marked as answer]
Thanks for the detailed answer. Let me make sure I fully grasped the concept. At first, I thought that when it only shows snapshot, that would be a restic backup triggered outside of backrest, so backrest didn't capture any of the (related) output, but backrest is the only tool using this repo and I guess this first assumption was wrong.


My next try to understand what you are trying to explain was that on this schedule of the plan, the backup didn't succeed hence no forget was triggered, and no backup details were captured, only showing the snapshot? 
This seems to be the case, as I found that covered in the docs here: https://garethgeorge.github.io/backrest/docs/operations 

> If the backup completed successfully, Backrest triggers followup operations




As far as I understand, plans in backrest have a schedule. 
Backrest uses this schedule to trigger `restic backup` and captures the (related) output. 

Example:
![image](https://github.com/garethgeorge/backrest/assets/837902/d02e31ba-24b8-4d0f-ab59-ffd9b18961f2)


But then why does the backup the day before only show snapshot, without any backup? This was also triggered by backrest as far as I can tell by looking at the date/time. Also, if it failed, why can I click on "Browse and restore files in backup" ?
![image](https://github.com/garethgeorge/backrest/assets/837902/7335f601-ae82-4c25-8856-e210be0844cd)

**Reply by garethgeorge** [reply]
Ah, I follow your source of confusion here seeing that you now have "Snapshots" in the top level view. This can happen when backrest loses association between a snapshot and the operations it ran to create it. This is common when

 * importing externally created snapshots
 * if the database has been reset
 * OR if a snapshot's ID changed e.g. by retagging it

From one of your other questions re: the 1.1.0 migration I suspect you may have run the migration command to retag your snapshots?

**Reply by ovizii** [reply]
> * importing externally created snapshots

I did that when I moved from autorestic to backrest
> From one of your other questions re: the 1.1.0 migration I suspect you may have run the migration command to retag your snapshots?

I did indeed follow the migration advice and retagged snapshots.


Furthermore, I am at ease now that we figured out the confusion. I assume this is only a cosmetic issue and will fade away, as backrest is now in full control of my repository and all my backups I assume.
